nifi-nar-bundles/nifi-marklogic-bundle/nifi-marklogic-processors/src/main/java/com/marklogic/nifi/processor/PutMarkLogic.java

```diff
@@ -0,0 +1,382 @@
+    "MarkLogic Data Movement SDK (DMSDK)")
+@TriggerWhenEmpty
+public class PutMarkLogic extends AbstractMarkLogicProcessor {
+
+    class FlowFileInfo {
+        FlowFile flowFile;
+        ProcessSession session;
+        FlowFileInfo(FlowFile flowFile, ProcessSession session) {
+            this.flowFile = flowFile;
+            this.session = session;
+        }
+    }
+    private Map<String, FlowFileInfo> URIFlowFileMap = new HashMap<>();
+    public static final PropertyDescriptor COLLECTIONS = new PropertyDescriptor.Builder()
+        .name("Collections")
+        .displayName("Collections")
+        .description("Comma-delimited sequence of collections to add to each document")
+        .addValidator(NO_VALIDATION_VALIDATOR)
+        .build();
+
+    public static final PropertyDescriptor FORMAT = new PropertyDescriptor.Builder()
+        .name("Format")
+        .displayName("Format")
+        .description("Format for each document; if not specified, MarkLogic will determine the format" +
+            " based on the URI")
+        .allowableValues(Format.JSON.name(), Format.XML.name(), Format.TEXT.name(), Format.BINARY.name(), Format.UNKNOWN.name())
+        .addValidator(NO_VALIDATION_VALIDATOR)
+        .build();
+
+    public static final PropertyDescriptor JOB_ID = new PropertyDescriptor.Builder()
+        .name("Job ID")
+        .displayName("Job ID")
+        .description("ID for the WriteBatcher job")
+        .addValidator(NO_VALIDATION_VALIDATOR)
+        .build();
+
+    public static final PropertyDescriptor JOB_NAME = new PropertyDescriptor.Builder()
+        .name("Job Name")
+        .displayName("Job Name")
+        .description("Name for the WriteBatcher job")
+        .addValidator(NO_VALIDATION_VALIDATOR)
+        .build();
+
+    public static final PropertyDescriptor MIMETYPE = new PropertyDescriptor.Builder()
+        .name("MIME type")
+        .displayName("MIME type")
+        .description("MIME type for each document; if not specified, MarkLogic will determine the " +
+            "MIME type based on the URI")
+        .addValidator(NO_VALIDATION_VALIDATOR)
+        .build();
+
+    public static final PropertyDescriptor PERMISSIONS = new PropertyDescriptor.Builder()
+        .name("Permissions")
+        .displayName("Permissions")
+        .defaultValue("rest-reader,read,rest-writer,update")
+        .description("Comma-delimited sequence of permissions - role1, capability1, role2, " +
+            "capability2 - to add to each document")
+        .addValidator(NO_VALIDATION_VALIDATOR)
+        .build();
+
+    public static final PropertyDescriptor TEMPORAL_COLLECTION = new PropertyDescriptor.Builder()
+        .name("Temporal collection")
+        .displayName("Temporal collection")
+        .description("The temporal collection to use for a temporal document insert")
+        .addValidator(NO_VALIDATION_VALIDATOR)
```

Comment:
This should use StandardValidators.NON_BLANK_VALIDATOR if you need it to be present. If not, use Validator.VALID.